File: js/js_css_generators/length.js
import { choice } from './util.js';
import { generate as generateNumber } from './number.js';

const suffixes = [
  "cap",
  "ch",
  "em",
  "ex",
  "ic",
  "lh",
  "rem",
  "rlh",
  "vh",
  "vw",
  "vi",
  "vb",
  "vmin",
  "vmax",
  "px",
  "cm",
  "mm",
  "Q",
  "in",
  "pc",
  "pt",
  "mozmm",
  "",
];

function generateSuffix() {
  return choice(suffixes);
}

export function generate() {
  const number = generateNumber();
  const suffix = generateSuffix();
  return `${number}${suffix}`;
}

File: src/lqc/generate/css/util/length.py
from random import choice
import re
from lqc.generate.css.util.number import generate as generate_number

suffixes = [
    "cap",
    "ch",
    "em",
    "ex",
    "ic",
    "lh",
    "rem",
    "rlh",
    "vh",
    "vw",
    "vi",
    "vb",
    "vmin",
    "vmax",
    "px",
    "cm",
    "mm",
    "Q",
    "in",
    "pc",
    "pt",
    "mozmm",
    "",
]


def generate_suffix():
    return choice(suffixes)


def generate():
    number = generate_number()
    suffix = generate_suffix()
    return f"{number}{suffix}"


def matches_length_pattern(style_string):
    LENGTH_RE = r'^([+-]*)([0-9]+)([a-zA-Z]+)$' # Matches numbers followed by letters e.g "100px", "2000"
    match = re.match(LENGTH_RE, style_string)
    if match is None:
        return False
    number_prefix, number, suffix = match.groups()
    return suffix in suffixes

    
